synapse/config/repository.py

```diff
@@ -62,18 +70,28 @@ def read_config(self, config):
         self.thumbnail_requirements = parse_thumbnail_requirements(
             config["thumbnail_sizes"]
         )
-        self.url_preview_enabled = config["url_preview_enabled"]
+        self.url_preview_enabled = config.get("url_preview_enabled", False)
         if self.url_preview_enabled:
             try:
+                import lxml
+                lxml  # To stop unused lint.
+            except ImportError:
+                raise ConfigError(MISSING_LXML)
```

Comment:
Do these ConfigError exceptions actually get raised anywhere meaningful? When I tried it this way before, there was no logging set up whilst the config was being parsed, so they got swallowed.